### PR TITLE
feat: Dashboard Filter Support

### PR DESCRIPTION
**Pull request by ttannis**
### Summary of Changes

This PR adds support for dashboard filters.

#### Python 
1. Updated the dashboard endpoint to include the same logic as the table logic that handles filters. This logic can be consolidated/shared in the future with a refactor but that was not a target goal for this PR.
2. Update the search helpers for filters to include support for dashboards.

#### React App
1. Added dashboard related options to the default configuration.
2. Added dashboards to the application filter state. The code that handles filtering is already written, and just needed these changes + endpoint update to work.
3. Update the `TagInfo` component so that when we click on tag it will execute a filter search across both tables and dashboards. This required updating the `updateSearchState` saga -- the flow is that clicking on a tag updates the filter state for tables and dashboards, then executes a search. 
4. Cleanup: Un-nested some test files.
5. Cleanup: Fixed and alignment issue in the `SearchFilter` UI. 

### Tests

Updates tests according to changes. 

### Documentation

N/A  for this PR but more details will be called out in the release notes. 

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes, including screenshots of any UI changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/components/SearchPage/SearchFilter/styles.scss [hunk 1]
```diff
@@ -15,6 +15,10 @@
       button {
         margin: auto 4px;
       }
+
+      label {
+        margin-bottom: 0;
```
Comment (ttannis) [inline amundsen_application/static/js/components/SearchPage/SearchFilter/styles.scss:20]: Fixes this misalignment issue:

image

**Comment by ttannis** [comment]
@Golodhros feel free to ask questions if you need any more context.

File: amundsen_application/static/js/components/SearchPage/SearchFilter/index.tsx [hunk 2]
```diff
@@ -73,9 +73,11 @@ export const mapStateToProps = (state: GlobalState) => {
         type: categoryConfig.type,
       };
       if (categoryConfig.type === FilterType.CHECKBOX_SELECT) {
-        section['options'] = categoryConfig.options.map((option) => {
-          return { value: option.value, label: option.displayName };
-        });
+        (section as CheckboxFilterSection).options = categoryConfig.options.map(
+          (option) => {
```
Comment (Golodhros) [inline amundsen_application/static/js/components/SearchPage/SearchFilter/index.tsx:77]: Nit: I would destructure the option's values here.

File: amundsen_application/static/js/ducks/search/api/v0.ts [hunk 3]
```diff
@@ -42,16 +42,15 @@ export function searchResource(
...
+    (resource === ResourceType.dashboard && !indexDashboardsEnabled()) ||
     (resource === ResourceType.user &&
       (!indexUsersEnabled() || term.length === 0))
   ) {
     return Promise.resolve({});
   }
 
   /* Note: This logic must exist until query string endpoints are created for all resources */
-  if (resource === ResourceType.table) {
+  if (resource === ResourceType.table || resource === ResourceType.dashboard) {
```
Comment (feng-tao) [inline amundsen_application/static/js/ducks/search/api/v0.ts:53]: 101qq: is this the main line to handle search filter for both dashboard and table? Per my read, we don't have any special logic to handle table filter vs dashboard filter in JS code?

Comment (ttannis) [reply amundsen_application/static/js/ducks/search/api/v0.ts:53]: Not particularly in the JS code. The filter configuration sets up the specific filter categories, and the application keeps track of the state of each one based on what the user has entered. This logic cares about what kind of filter it is (e.g. single input or multi-select checkbox) but  not the actual content. 

Then on line 55 that information will either get sent to `/api/search/v0/table` or `/api/search/v0/dashboard` in the Flask layer.

**Pull request merged**